packages/react-aria/src/hooks/useAriaButton.ts

```diff
@@ -0,0 +1,94 @@
+import * as React from 'react';
+import { ObjectShorthandProps, useEventCallback } from '@fluentui/react-utilities';
+
+export function useARIAButton(shorthand: ObjectShorthandProps<React.ButtonHTMLAttributes<HTMLElement>>) {
+  if (!shorthand.hasOwnProperty('as')) {
```

Comment:
**chore:** can we return a value from the hook (in this case probably just the mutated shorthand)? I mean there are so many  mutations happening within the hook.... to follow best practices hooks should return always some API.

Reply:
I don't think this should be a hook, our component hooks are a bit of a special case because they _**might**_ at some point use react hooks in the future

Reply:
Well. to be fair the first implementation of this hook was actually pure, but I've opted for this approach since it's more performant, WDYT?!

Reply:
> more performant

this is indeed tricky - memoizing stuff === bigger memory impact, there is no silver bullet